mayastor/tests/lock_lba_range.rs

```diff
@@ -0,0 +1,269 @@
+    });
+}
+
+fn get_disk(number: u64) -> String {
+    format!("/tmp/disk{}.img", number)
+}
+
+fn get_dev(number: u64) -> String {
+    format!("aio://{}?blk_size=512", get_disk(number))
+}
+
+async fn create_nexus() {
+    let mut ch = Vec::new();
+    for i in 0 .. NUM_NEXUS_CHILDREN {
+        ch.push(get_dev(i));
+    }
+
+    nexus_create(NEXUS_NAME, NEXUS_SIZE, None, &ch)
+        .await
+        .unwrap();
+}
+
+fn get_shareable_ctx(offset: u64, len: u64) -> Arc<Mutex<RangeContext>> {
+    Arc::new(Mutex::new(RangeContext::new(offset, len)))
+}
+
+fn get_shareable_nexus_channel() -> Arc<Mutex<IoChannel>> {
+    let nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    Arc::new(Mutex::new(nexus.get_channel().unwrap()))
+}
+
+async fn lock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.lock_lba_range(ctx, ch).await;
+}
+
+async fn unlock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.unlock_lba_range(ctx, ch).await;
+}
+
+#[test]
+// Test acquiring and releasing a lock.
+fn lock_unlock() {
+    test_ini();
+    Reactor::block_on(async {
+        let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+        let mut ctx = RangeContext::new(1, 5);
+        let ch = nexus.get_channel().unwrap();
+        let _ = nexus.lock_lba_range(&mut ctx, &ch).await;
+        let _ = nexus.unlock_lba_range(&mut ctx, &ch).await;
+    });
+    test_fini();
+}
+
+#[test]
+// Test that an error is received when the lock/unlock contexts don't match
+fn lock_unlock_different_context() {
+    test_ini();
+    Reactor::block_on(async {
+        let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+
+        let mut ctx = RangeContext::new(1, 5);
+        let ch = nexus.get_channel().unwrap();
+        let _ = nexus.lock_lba_range(&mut ctx, &ch).await;
+
+        let mut ctx1 = RangeContext::new(1, 5);
+        let ch1 = nexus.get_channel().unwrap();
+        if nexus.unlock_lba_range(&mut ctx1, &ch1).await.is_ok() {
+            panic!("Shouldn't be able to unlock with a different context");
+        }
+    });
+    test_fini();
+}
+
+#[test]
+// Test taking out multiple locks on an overlapping block range.
+// The second lock should only succeeded after the first lock is released.
+fn multiple_locks() {
+    test_ini();
+    {
+        let reactor = Reactors::current();
+
+        // First lock
+        let (s, r) = unbounded::<()>();
+        let ctx = get_shareable_ctx(1, 10);
+        let ctx_clone = Arc::clone(&ctx);
+        let ch = get_shareable_nexus_channel();
+        let ch_clone = Arc::clone(&ch);
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx_clone.lock().unwrap(),
+                &ch_clone.lock().unwrap(),
+            )
+            .await;
+            let _ = s.send(());
```

Comment:
if you dont care for the result, its best to just unwrap :)

Reply:
Done